core/src/main/java/haveno/core/account/AccountService.java

```diff
@@ -0,0 +1,190 @@
+			log.error("Accounts have not been initialized " + e.getMessage());
+			return false;
+		}
+		return true;
+	}
+	
+	public void openAccount(String password) throws AccountException {
+		if (!accountExists()) { 
+			throw new AccountException("Account does not exists");
+		}
+		config.setHavenoWalletPassword(password);
+		ObjectProperty<Throwable> accountServiceException = new SimpleObjectProperty<>();
+		this.walletsSetup.initialize(null,               
+				() -> {
+	                // We only check one wallet as we apply encryption to all or none
+//	                if (walletsManager.areWalletsEncrypted() && !coreContext.isApiUser()) {
+//	                    walletPasswordHandler.run();
+//	                } else {
+//	                    if (isSpvResyncRequested && !coreContext.isApiUser()) {
+//	                        if (showFirstPopupIfResyncSPVRequestedHandler != null)
+//	                            showFirstPopupIfResyncSPVRequestedHandler.run();
+//	                    } else {
+//	                        walletInitializedHandler.run();
+//	                    }
+//	                }
+	            },
+	            exception -> {
+//	                if (exception instanceof InvalidHostException && showPopupIfInvalidBtcConfigHandler != null) {
+//	                    showPopupIfInvalidBtcConfigHandler.run();
+//	                } else {
+	            	accountServiceException.set(exception);
+//	                }
+	            });
+	}
+	
+	public void closeAccount() throws AccountException {
+		if (!isAccountOpen()) { 
+			throw new AccountException("Account is not open!");
+		}
+		this.walletsSetup.shutDown();
+		walletsInitialized = false;
+	}
+	
+	public BufferedInputStream backupAccount() throws AccountException {
+		if (!accountExists()) { 
+			throw new AccountException("Account does not exists");
+		}
+		var zipName = config.appDataDir.getAbsolutePath() + "/xmr_stagenet.zip";
+		File zipFile = new File(zipName);
+		if (zipFile.exists())
+			zipFile.delete();
+		walletsSetup.clearBackups();
+		Zip.compressDirectory(config.appDataDir.getAbsolutePath(), zipName);
+		
+      	FileInputStream fis = null;
+		try {
+			fis = new FileInputStream(zipName);
+		} catch (FileNotFoundException e) {
+			log.error("Could not load" + zipName + e.getMessage());
+			e.printStackTrace();
+		}
+		return new BufferedInputStream(fis);		
+	}
+	
+	public void deleteAccount() {
+		try {
+			this.walletsSetup.shutDown();
+			var dir = new File(config.appDataDir(), config.baseCurrencyNetwork().name().toLowerCase());
+			FileUtil.deleteDirectory(dir);
+			walletsInitialized = false;
+		} catch (IOException e) {
+			log.error("Could not delete directory " + e.getMessage());
+			e.printStackTrace();
+		}
+				
+	}
+	
+	public void restoreAccount() throws AccountException {
+		if (accountExists()) { 
+			throw new AccountException("Account already exists!");
+		}
+	}
+	
+	public void changePassword(String password) throws AccountException {
```

Comment:
This will not change the passwords as implemented.

Should call `CoreWalletService.setWalletPassword()` which needs updated to call `XmrWalletService.setWalletPasswords(oldPassword, newPassword)` which for each main wallet and multisig wallets, calls wallet.changePassword().

Reply:
CoreWalletsService is not public, you want it changed?

Reply:
It can be injected into the constructor of other services?

Reply:
was an issue with a package call haveno, it had no access to CoreWalletService. i reverted this